### PR TITLE
New package: lego-2.4.0

### LINKED ISSUE #10588: Package Request: Lego
URL: https://go-acme.github.io/lego/
Description: 
```
Let's Encrypt client and ACME library written in Go
```
Source: https://github.com/go-acme/lego

Reason: Seems to be the best up-to-date acme-client right now.

### Features

- Register with CA
- Obtain certificates, both from scratch or with an existing CSR
- Renew certificates
- Revoke certificates
- Robust implementation of all ACME challenges
  - HTTP (http-01)
  - DNS (dns-01)
  - TLS (tls-alpn-01)
- SAN certificate support
- Comes with multiple optional [DNS providers](https://go-acme.github.io/lego/dns)
- [Custom challenge solvers](https://go-acme.github.io/lego/usage/library/writing-a-challenge-solver/)
- Certificate bundling
- OCSP helper function

Note: `lego introduced support for ACME v2 in [v1.0.0](https://github.com/go-acme/lego/releases/tag/v1.0.0).`

### PR DESCRIPTION
**Pull request by Anachron**
Fixes #10588

### CODE REVIEW

File: srcpkgs/lego/template [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'lego'
+pkgname=lego
+version=2.4.0
+revision=1
+archs="i686 x86_64 armv7 aarch64"
```
Comment (maxice8) [inline srcpkgs/lego/template]: uh, shouldn't it also be in musl ?

Comment (jnbr) [reply srcpkgs/lego/template]: if there is no specific reason, just drop this line to enable all architectures

**Comment by maxice8** [comment]
On Mon, Apr 08, 2019 at 03:19:14AM -0700, Johannes wrote:
> jnbr commented on this pull request.
> 
> 
> 
> > @@ -0,0 +1,18 @@
> +# Template file for 'lego'
> +pkgname=lego
> +version=2.4.0
> +revision=1
> +archs="i686 x86_64 armv7 aarch64"
> 
> if there is no specific reason, just drop this line to enable all architectures
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10692#discussion_r272976601

Yes, he said on IRC he is dropping it.

**Comment by Anachron** [comment]
All is good now.

**Pull request merged**